examples/4337-gas-metering/pimlico/pimlico.ts

```diff
@@ -1,147 +1,163 @@
+}
+
+// Extract Signer from Private Key.
 const signer = privateKeyToAccount(privateKey as Hash)
 console.log('Signer Extracted from Private Key.')
 
-let bundlerClient
-let publicClient
-let pimlicoPaymasterClient
-if (chain == 'goerli') {
-  bundlerClient = createClient({
-    transport: http(`https://api.pimlico.io/v1/${chain}/rpc?apikey=${apiKey}`),
-    chain: goerli,
-  })
-    .extend(bundlerActions)
-    .extend(pimlicoBundlerActions)
-
-  publicClient = createPublicClient({
-    transport: http(rpcURL),
-    chain: goerli,
-  })
-
-  pimlicoPaymasterClient = createClient({
-    transport: http(`https://api.pimlico.io/v2/${chain}/rpc?apikey=${apiKey}`),
-    chain: goerli,
-  }).extend(pimlicoPaymasterActions)
-} else if (chain == 'mumbai') {
-  bundlerClient = createClient({
-    transport: http(`https://api.pimlico.io/v1/${chain}/rpc?apikey=${apiKey}`),
-    chain: polygonMumbai,
-  })
-    .extend(bundlerActions)
-    .extend(pimlicoBundlerActions)
+// Create a public, bundler and paymaster Client for the Chain.
+const viemChain = chain == 'sepolia' ? sepolia : baseSepolia
+const bundlerClient = createClient({
+  transport: http(`https://api.pimlico.io/v2/${chain}/rpc?apikey=${apiKey}`),
+  chain: viemChain,
+})
+  .extend(bundlerActions(ENTRYPOINT_ADDRESS_V07))
+  .extend(pimlicoBundlerActions(ENTRYPOINT_ADDRESS_V07))
 
-  publicClient = createPublicClient({
-    transport: http(rpcURL),
-    chain: polygonMumbai,
-  })
+const publicClient = createPublicClient({
+  transport: http(rpcURL),
+  chain: viemChain,
+})
 
-  pimlicoPaymasterClient = createClient({
-    transport: http(`https://api.pimlico.io/v2/${chain}/rpc?apikey=${apiKey}`),
-    chain: polygonMumbai,
-  }).extend(pimlicoPaymasterActions)
-} else {
-  throw new Error('Current code only support limited networks. Please make required changes if you want to use custom network.')
-}
+const pimlicoPaymasterClient = bundlerClient.extend(pimlicoPaymasterActions(ENTRYPOINT_ADDRESS_V07))
 
+// Get Safe Init Code.
 const initCode = await getAccountInitCode({
   owner: signer.address,
-  addModuleLibAddress: chainAddresses.ADD_MODULES_LIB_ADDRESS,
+  safeModuleSetupAddress: chainAddresses.SAFE_MODULE_SETUP_ADDRESS,
   safe4337ModuleAddress: chainAddresses.SAFE_4337_MODULE_ADDRESS,
-  safeProxyFactoryAddress: chainAddresses.SAFE_PROXY_FACTORY_ADDRESS,
   safeSingletonAddress: chainAddresses.SAFE_SINGLETON_ADDRESS,
   saltNonce: saltNonce,
   multiSendAddress: multiSendAddress,
-  erc20TokenAddress: usdcTokenAddress,
-  paymasterAddress: erc20PaymasterAddress,
+  erc20TokenAddress: transactionType === UserOperationType.ERC20Paymaster ? usdcTokenAddress : zeroAddress,
+  paymasterAddress: transactionType === UserOperationType.ERC20Paymaster ? erc20PaymasterAddress : zeroAddress,
 })
 console.log('\nInit Code Created.')
 
+// Get Safe Address Counterfactually.
 const senderAddress = await getAccountAddress({
```

Comment:
Potential improvement: both `getAccountInitCode` and `getAccountAddress` compute the `initializer`, which is error-prone. If you mistakenly pass different data to one of them, they won't match.